lib/nodes/decorators/Retry.ts

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+import {ResultCodes} from '../../utils/ResultCodes';
+import {Base} from '../Base';
+import {RepeatWhenNTimes} from './RepeatWhenNTimes';
+
+export class Retry<State> extends RepeatWhenNTimes<State> {
+
+	constructor(child: Base<State>, numRetries: number) {
```

Comment:
How is this different than RepeatOnResult with rc.FAILURE?

Reply:
It has numRetries.  Maybe we should add this to `RepeatOnResult` instead?

Reply:
I think I would rather add this new class with new functionality (numRetries) than modify RepeatOnResult

Reply:
I wanted to use 1 common base class for all of the repeat functionality, rather than create a completely separate class.